crates/hyprstream/src/cli/git_handlers.rs

```diff
@@ -1014,21 +946,20 @@ pub async fn handle_infer(
         info!("Generated {} tokens", result.split_whitespace().count());
     } else {
         // Generate all at once using streaming with collection
-        let mut response = String::new();
+        let response = std::sync::Arc::new(std::sync::Mutex::new(String::new()));
```

Comment:
Using `std::sync::Mutex` inside async contexts can block the executor. Should probably use `tokio::sync::Mutex` instead. 

`let response = std::sync::Arc::new(AsyncMutex::new(String::new()));`

If you make this change, the other comments below go with it, if not, ignore the other comments.